pylint/message/message_handler_mix_in.py

```diff
@@ -369,7 +359,7 @@ def add_one_message(  # type: ignore # MessagesHandlerMixIn is always mixed with
             Message(
                 message_definition.msgid,
                 message_definition.symbol,
-                (abspath, path, module, obj, line or 1, col_offset or 0),
+                (abspath, path, module, obj, line or 1, col_offset or 0),  # type: ignore
```

Comment:
The types of `abspath`, `path` and `module` include `Any` and `None`. We need some overloads in `astroid` to fix this. I thought it best to add a `type: ignore` instead of adding multiple `isinstance` checks